---
title: "multimedia"
output: md_document
---

```{r, echo = FALSE}
knitr::knit_hooks$set(output = multimedia::ansi_aware_handler)
knitr::opts_chunk$set(fig.path = "man/figures/README-")
options(crayon.enabled = TRUE)
set.seed(20240313)
```

# multimedia <img src="man/figures/logo.png" align="right" width="200" alt=""/>

<!-- badges: start -->
[![R-CMD-check](https://github.com/krisrs1128/multimedia/actions/workflows/R-CMD-check.yaml/badge.svg)](https://github.com/krisrs1128/multimedia/actions/workflows/R-CMD-check.yaml)
[![codecov](https://codecov.io/github/krisrs1128/multimedia/graph/badge.svg?token=XB865GLQ8B)](https://app.codecov.io/github/krisrs1128/multimedia/)
[![Launch binder](https://mybinder.org/badge_logo.svg)](https://mybinder.org/v2/gh/krisrs1128/multimedia/main/)
<!-- badges: end -->

`multimedia` is an R package for **multi**modal **media**tion analysis of
microbiome data. It streamlines data curation, mediation and outcome model
specification, and statistical inference for direct and indirect effects. By
defining causal relationships across data modalities, it can support principled
data integration. You can read more about the package in our preprint:

[H. Jiang, X. Miao, M. W. Thairu, M. Beebe, D. W. Grupe, R. J. Davidson, J. Handelsman, and K. Sankaran (2024). multimedia: Multimodal Mediation Analysis of Microbiome Data.](https://www.biorxiv.org/content/10.1101/2024.03.27.587024v1)

The preprint describes the scientific context and interpretation for two of the
vignettes in this package. One gives a multi-omics analysis of IBD, and the
other describes how to simultaneously model 16S profiles and survey responses in
a mindfulness intervention study.

<center>
  <img src="man/figures/overview_figure-extended.png" width=400/>
</center>

## Installation

You can install the development version from
[GitHub](https://github.com/krisrs1128/multimedia) with:

``` r
# install.packages("devtools")
devtools::install_github("krisrs1128/multimedia")
```

## Example

Here is a simple example of estimating direct and indirect effects. 
The data are randomly generated (no real effects), but we imagine that
the ASV columns mediate the relationship between `treatment` and `PHQ`.  This is
mimics the possibility that the microbiome (ASV = Amplicon Sequence Variant)
mediates the relationship between a treatment and depression (PHQ = Patient
Health Questionnaire). You can find more details in the `random.Rmd` vignette.

The original data here are a `SummarizedExperiment`. The package can also take
phyloseq objects and data.frames.

```{r, message = FALSE}
library(multimedia)
demo_joy()
```

Next, we specify which columns are the treatment, mediators, and outcomes.
Notice that we can use `tidyselect` syntax to match multiple columns.

```{r}
exper <- mediation_data(demo_joy(), "PHQ", "treatment", starts_with("ASV"))
exper
```

Next, we fit all mediation analysis components and estimate effects. By default,
the package uses linear models -- see the vignettes for examples using sparse
regression, random forests, and bayesian hierarchical models instead.

```{r}
model <- multimedia(exper) |>
    estimate(exper)
model
```

In any mediation analysis, there are several types of effects that could be
interesting, each corresponding to different ways of traveling from the
treatment to the outcome in the mediation analysis causal graph. In the block
below, `direct_effect` captures treatment effects that bypass the microbiome;
`indirect_effect` are effects that are mediated by ASV relative abundances.
Since this example uses a linear model, the effects are identical for the two
indirect settings.

```{r}
direct_effect(model, exper)
indirect_overall(model, exper)
```

The package also includes helpers to visualize and perform inference on these
effects. For example, 

```{r, fig.height = 3, fig.width = 6, dpi = 300, out.width=500, fig.align="center", message = FALSE}
boot <- bootstrap(model, exper, c(direct = direct_effect))

library(ggplot2)
ggplot(boot$direct) +
    geom_histogram(aes(direct_effect), bins = 20) +
    scale_y_continuous(expand = c(0, 0)) +
    theme_classic() +
    labs(x = "Direct Effect", y = "Frequency", title = "Bootstrap Distribution")
```

If we want to use a different type of model, we can just modify the original
`multimedia` specification. Below we use a sparse regression model, which
correctly recovers that the direct effects are 0.

```{r}
multimedia(exper, glmnet_model(lambda = .1)) |>
    estimate(exper) |>
    direct_effect()
```

## Help

We welcome questions and comments about the package either through
[github](https://github.com/krisrs1128/multimedia/issues) or [email](mailto:ksankaran@wisc.edu).

```{r, eval = FALSE, include = FALSE}
library(hexSticker)
library(showtext)
img <- "~/Downloads/source.jpeg"
font_add_google("Dosis", "dosis")
sticker(img, package = "multimedia", h_color = "#a92645", h_fill = "#f7f7f7", p_size = 35, p_y = 1.4, s_x = 1, s_y = .75, s_width = .47, p_color = "#0c0c0c", dpi = 500, p_family = "dosis")
usethis::use_logo("multimedia.png")
```

```{r}
sessionInfo()
```
