<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Multimodal Mediation Analysis • multimedia</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="apple-touch-icon-60x60.png">
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="deps/headroom-0.11.0/headroom.min.js"></script><script src="deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="deps/search-1.0.0/fuse.min.js"></script><script src="deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Multimodal Mediation Analysis">
<meta name="description" content="Multimodal mediation analysis is an emerging problem in microbiome data analysis. Multimedia make advanced mediation analysis techniques easy to use, ensuring that all statistical components are transparent and adaptable to specific problem contexts. The package provides a uniform interface to direct and indirect effect estimation, synthetic null hypothesis testing, and bootstrap confidence interval construction. More details are available in Jiang et al. (2024) &lt;doi:10.1101/2024.03.27.587024&gt;.">
<meta property="og:description" content="Multimodal mediation analysis is an emerging problem in microbiome data analysis. Multimedia make advanced mediation analysis techniques easy to use, ensuring that all statistical components are transparent and adaptable to specific problem contexts. The package provides a uniform interface to direct and indirect effect estimation, synthetic null hypothesis testing, and bootstrap confidence interval construction. More details are available in Jiang et al. (2024) &lt;doi:10.1101/2024.03.27.587024&gt;.">
<meta property="og:image" content="https://krisrs1128.github.io/multimedia/logo.png">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="index.html">multimedia</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.2.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="articles/IBD.html">Microbiome - Metabolome Mediation Analysis</a></li>
    <li><a class="dropdown-item" href="articles/illustration.html">Illustration with Nonlinear Effects</a></li>
    <li><a class="dropdown-item" href="articles/mindfulness.html">The Mindfulness Study</a></li>
    <li><a class="dropdown-item" href="articles/random.html">Quick Start with Random Data</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/krisrs1128/multimedia/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header">
<img src="logo.png" class="logo" alt=""><h1 id="multimedia-">multimedia <a class="anchor" aria-label="anchor" href="#multimedia-"></a>
</h1>
</div>
<!-- badges: start -->

<p><code>multimedia</code> is an R package for <strong>multi</strong>modal <strong>media</strong>tion analysis of microbiome data. It streamlines data curation, mediation and outcome model specification, and statistical inference for direct and indirect effects. By defining causal relationships across data modalities, it can support principled data integration. You can read more about the package in our preprint:</p>
<p><a href="https://www.biorxiv.org/content/10.1101/2024.03.27.587024v1" class="external-link">H. Jiang, X. Miao, M. W. Thairu, M. Beebe, D. W. Grupe, R. J. Davidson, J. Handelsman, and K. Sankaran (2024). multimedia: Multimodal Mediation Analysis of Microbiome Data.</a></p>
<p>The preprint describes the scientific context and interpretation for two of the vignettes in this package. One gives a multi-omics analysis of IBD, and the other describes how to simultaneously model 16S profiles and survey responses in a mindfulness intervention study.</p>
<center>
<img src="reference/figures/overview_figure-extended.png" width="400/">
</center>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>You can install the development version from <a href="https://github.com/krisrs1128/multimedia" class="external-link">GitHub</a> with:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="co"># install.packages("devtools")</span></span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">"krisrs1128/multimedia"</span>)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="example">Example<a class="anchor" aria-label="anchor" href="#example"></a>
</h2>
<p>Here is a simple example of estimating direct and indirect effects. The data are randomly generated (no real effects), but we imagine that the ASV columns mediate the relationship between <code>treatment</code> and <code>PHQ</code>. This is mimics the possibility that the microbiome (ASV = Amplicon Sequence Variant) mediates the relationship between a treatment and depression (PHQ = Patient Health Questionnaire). You can find more details in the <code>random.Rmd</code> vignette.</p>
<p>The original data here are a <code>SummarizedExperiment</code>. The package can also take phyloseq objects and data.frames.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="fu">library</span>(multimedia)</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="fu">demo_joy</span>()</span></code></pre></div>
<pre class="r-output"><code><span><span class="co">## class: SummarizedExperiment</span></span>
<span><span class="co">## dim: 5 100</span></span>
<span><span class="co">## metadata(0):</span></span>
<span><span class="co">## assays(1): counts</span></span>
<span><span class="co">## rownames(5): ASV1 ASV2 ASV3 ASV4 ASV5</span></span>
<span><span class="co">## rowData names(0):</span></span>
<span><span class="co">## colnames: NULL</span></span>
<span><span class="co">## colData names(2): treatment PHQ</span></span></code></pre>
<p>Next, we specify which columns are the treatment, mediators, and outcomes. Notice that we can use <code>tidyselect</code> syntax to match multiple columns.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a>exper <span class="ot">&lt;-</span> <span class="fu">mediation_data</span>(<span class="fu">demo_joy</span>(), <span class="st">"PHQ"</span>, <span class="st">"treatment"</span>, <span class="fu">starts_with</span>(<span class="st">"ASV"</span>))</span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a>exper</span></code></pre></div>
<pre class="r-output"><code><span><span class="co">## [Mediation Data]</span></span>
<span><span class="co">## 100 samples with measurements for,</span></span>
<span><span class="co">## 1 treatment: treatment</span></span>
<span><span class="co">## 5 mediators: ASV1, ASV2, ...</span></span>
<span><span class="co">## 1 outcome: PHQ</span></span></code></pre>
<p>Next, we fit all mediation analysis components and estimate effects. By default, the package uses linear models – see the vignettes for examples using sparse regression, random forests, and bayesian hierarchical models instead.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a>model <span class="ot">&lt;-</span> <span class="fu">multimedia</span>(exper) <span class="sc">|&gt;</span></span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a>    <span class="fu">estimate</span>(exper)</span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a>model</span></code></pre></div>
<pre class="r-output"><code><span><span class="co">## [Multimedia Analysis]</span></span>
<span><span class="co">## Treatments: treatment</span></span>
<span><span class="co">## Outcomes: PHQ</span></span>
<span><span class="co">## Mediators: ASV1, ASV2, ...</span></span>
<span><span class="co">##</span></span>
<span><span class="co">## [Models]</span></span>
<span><span class="co">## mediation: A fitted lm_model().</span></span>
<span><span class="co">## outcome: A fitted lm_model().</span></span></code></pre>
<p>In any mediation analysis, there are several types of effects that could be interesting, each corresponding to different ways of traveling from the treatment to the outcome in the mediation analysis causal graph. In the block below, <code>direct_effect</code> captures treatment effects that bypass the microbiome; <code>indirect_effect</code> are effects that are mediated by ASV relative abundances. Since this example uses a linear model, the effects are identical for the two indirect settings.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="fu">direct_effect</span>(model, exper)</span></code></pre></div>
<pre class="r-output"><code><span><span class="co">##   outcome indirect_setting            contrast direct_effect</span></span>
<span><span class="co">## 1     PHQ          Control Control - Treatment    0.09314376</span></span>
<span><span class="co">## 2     PHQ        Treatment Control - Treatment    0.09314376</span></span></code></pre>
<div class="sourceCode" id="cb6"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="fu">indirect_overall</span>(model, exper)</span></code></pre></div>
<pre class="r-output"><code><span><span class="co">##   outcome direct_setting            contrast indirect_effect</span></span>
<span><span class="co">## 1     PHQ        Control Control - Treatment      0.02256029</span></span>
<span><span class="co">## 2     PHQ      Treatment Control - Treatment      0.02256029</span></span></code></pre>
<p>The package also includes helpers to visualize and perform inference on these effects. For example,</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a>boot <span class="ot">&lt;-</span> <span class="fu">bootstrap</span>(model, exper, <span class="fu">c</span>(<span class="at">direct =</span> direct_effect))</span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a></span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a><span class="fu">ggplot</span>(boot<span class="sc">$</span>direct) <span class="sc">+</span></span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a>    <span class="fu">geom_histogram</span>(<span class="fu">aes</span>(direct_effect), <span class="at">bins =</span> <span class="dv">20</span>) <span class="sc">+</span></span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a>    <span class="fu">scale_y_continuous</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb7-7"><a href="#cb7-7" tabindex="-1"></a>    <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb7-8"><a href="#cb7-8" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Direct Effect"</span>, <span class="at">y =</span> <span class="st">"Frequency"</span>, <span class="at">title =</span> <span class="st">"Bootstrap Distribution"</span>)</span></code></pre></div>
<p><img src="reference/figures/README-unnamed-chunk-6-1.png" width="500" style="display: block; margin: auto;"></p>
<p>If we want to use a different type of model, we can just modify the original <code>multimedia</code> specification. Below we use a sparse regression model, which correctly recovers that the direct effects are 0.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="fu">multimedia</span>(exper, <span class="fu">glmnet_model</span>(<span class="at">lambda =</span> .<span class="dv">1</span>)) <span class="sc">|&gt;</span></span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a>    <span class="fu">estimate</span>(exper) <span class="sc">|&gt;</span></span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a>    <span class="fu">direct_effect</span>()</span></code></pre></div>
<pre class="r-output"><code><span><span class="co">##   outcome indirect_setting            contrast direct_effect</span></span>
<span><span class="co">## 1     PHQ          Control Control - Treatment             0</span></span>
<span><span class="co">## 2     PHQ        Treatment Control - Treatment             0</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="help">Help<a class="anchor" aria-label="anchor" href="#help"></a>
</h2>
<p>We welcome questions and comments about the package either through <a href="https://github.com/krisrs1128/multimedia/issues" class="external-link">github</a> or <a href="mailto:ksankaran@wisc.edu">email</a>.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre class="r-output"><code><span><span class="co">## R version 4.4.1 Patched (2024-08-21 r87049)</span></span>
<span><span class="co">## Platform: aarch64-apple-darwin20</span></span>
<span><span class="co">## Running under: macOS Sonoma 14.5</span></span>
<span><span class="co">##</span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib</span></span>
<span><span class="co">## LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0</span></span>
<span><span class="co">##</span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">##</span></span>
<span><span class="co">## time zone: America/Chicago</span></span>
<span><span class="co">## tzcode source: internal</span></span>
<span><span class="co">##</span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base</span></span>
<span><span class="co">##</span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] ggplot2_3.5.1     multimedia_0.2.0  tidyselect_1.2.1  ranger_0.16.0</span></span>
<span><span class="co">## [5] glmnetUtils_1.1.9 brms_2.21.0       Rcpp_1.0.13</span></span>
<span><span class="co">##</span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] gridExtra_2.3               inline_0.3.19</span></span>
<span><span class="co">##   [3] permute_0.9-7               sandwich_3.1-0</span></span>
<span><span class="co">##   [5] rlang_1.1.4                 magrittr_2.0.3</span></span>
<span><span class="co">##   [7] multcomp_1.4-26             ade4_1.7-22</span></span>
<span><span class="co">##   [9] matrixStats_1.4.0           compiler_4.4.1</span></span>
<span><span class="co">##  [11] mgcv_1.9-1                  loo_2.8.0</span></span>
<span><span class="co">##  [13] vctrs_0.6.5                 reshape2_1.4.4</span></span>
<span><span class="co">##  [15] stringr_1.5.1               pkgconfig_2.0.3</span></span>
<span><span class="co">##  [17] shape_1.4.6.1               crayon_1.5.3</span></span>
<span><span class="co">##  [19] fastmap_1.2.0               backports_1.5.0</span></span>
<span><span class="co">##  [21] XVector_0.45.0              labeling_0.4.3</span></span>
<span><span class="co">##  [23] utf8_1.2.4                  rmarkdown_2.28</span></span>
<span><span class="co">##  [25] UCSC.utils_1.1.0            purrr_1.0.2</span></span>
<span><span class="co">##  [27] xfun_0.47                   glmnet_4.1-8</span></span>
<span><span class="co">##  [29] zlibbioc_1.51.1             GenomeInfoDb_1.41.1</span></span>
<span><span class="co">##  [31] jsonlite_1.8.8              progress_1.2.3</span></span>
<span><span class="co">##  [33] biomformat_1.33.0           highr_0.11</span></span>
<span><span class="co">##  [35] rhdf5filters_1.17.0         DelayedArray_0.31.11</span></span>
<span><span class="co">##  [37] Rhdf5lib_1.27.0             prettyunits_1.2.0</span></span>
<span><span class="co">##  [39] cluster_2.1.6               parallel_4.4.1</span></span>
<span><span class="co">##  [41] R6_2.5.1                    stringi_1.8.4</span></span>
<span><span class="co">##  [43] StanHeaders_2.32.10         GenomicRanges_1.57.1</span></span>
<span><span class="co">##  [45] estimability_1.5.1          rstan_2.32.6</span></span>
<span><span class="co">##  [47] SummarizedExperiment_1.35.1 iterators_1.0.14</span></span>
<span><span class="co">##  [49] knitr_1.48                  zoo_1.8-12</span></span>
<span><span class="co">##  [51] IRanges_2.39.2              bayesplot_1.11.1</span></span>
<span><span class="co">##  [53] igraph_2.0.3                Matrix_1.7-0</span></span>
<span><span class="co">##  [55] splines_4.4.1               abind_1.4-5</span></span>
<span><span class="co">##  [57] yaml_2.3.10                 vegan_2.6-8</span></span>
<span><span class="co">##  [59] codetools_0.2-20            curl_5.2.2</span></span>
<span><span class="co">##  [61] pkgbuild_1.4.4              lattice_0.22-6</span></span>
<span><span class="co">##  [63] tibble_3.2.1                plyr_1.8.9</span></span>
<span><span class="co">##  [65] withr_3.0.1                 Biobase_2.65.1</span></span>
<span><span class="co">##  [67] bridgesampling_1.1-2        posterior_1.6.0</span></span>
<span><span class="co">##  [69] coda_0.19-4.1               evaluate_0.24.0</span></span>
<span><span class="co">##  [71] survival_3.7-0              RcppParallel_5.1.9</span></span>
<span><span class="co">##  [73] Biostrings_2.73.1           pillar_1.9.0</span></span>
<span><span class="co">##  [75] phyloseq_1.49.0             MatrixGenerics_1.17.0</span></span>
<span><span class="co">##  [77] tensorA_0.36.2.1            checkmate_2.3.2</span></span>
<span><span class="co">##  [79] foreach_1.5.2               stats4_4.4.1</span></span>
<span><span class="co">##  [81] distributional_0.4.0        generics_0.1.3</span></span>
<span><span class="co">##  [83] hms_1.1.3                   S4Vectors_0.43.2</span></span>
<span><span class="co">##  [85] rstantools_2.4.0            munsell_0.5.1</span></span>
<span><span class="co">##  [87] scales_1.3.0                xtable_1.8-4</span></span>
<span><span class="co">##  [89] glue_1.7.0                  emmeans_1.10.4</span></span>
<span><span class="co">##  [91] tools_4.4.1                 data.table_1.16.0</span></span>
<span><span class="co">##  [93] mvtnorm_1.3-1               tidygraph_1.3.1</span></span>
<span><span class="co">##  [95] rhdf5_2.49.0                grid_4.4.1</span></span>
<span><span class="co">##  [97] tidyr_1.3.1                 ape_5.8</span></span>
<span><span class="co">##  [99] QuickJSR_1.3.1              miniLNM_0.1.0</span></span>
<span><span class="co">## [101] colorspace_2.1-1            nlme_3.1-166</span></span>
<span><span class="co">## [103] formula.tools_1.7.1         GenomeInfoDbData_1.2.12</span></span>
<span><span class="co">## [105] patchwork_1.2.0             cli_3.6.3</span></span>
<span><span class="co">## [107] fansi_1.0.6                 S4Arrays_1.5.7</span></span>
<span><span class="co">## [109] Brobdingnag_1.2-9           dplyr_1.1.4</span></span>
<span><span class="co">## [111] V8_5.0.0                    gtable_0.3.5</span></span>
<span><span class="co">## [113] digest_0.6.37               operator.tools_1.6.3</span></span>
<span><span class="co">## [115] BiocGenerics_0.51.1         SparseArray_1.5.31</span></span>
<span><span class="co">## [117] TH.data_1.1-2               farver_2.1.2</span></span>
<span><span class="co">## [119] multtest_2.61.0             htmltools_0.5.8.1</span></span>
<span><span class="co">## [121] lifecycle_1.0.4             httr_1.4.7</span></span>
<span><span class="co">## [123] MASS_7.3-61</span></span></code></pre>
</div>
</div>
  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/krisrs1128/multimedia/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/krisrs1128/multimedia/issues/" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small>CC0</small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing multimedia</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Kris Sankaran <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-9415-1971" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Hanying Jiang <br><small class="roles"> Author </small>  </li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/krisrs1128/multimedia/actions/workflows/R-CMD-check.yaml" class="external-link"><img src="https://github.com/krisrs1128/multimedia/actions/workflows/R-CMD-check.yaml/badge.svg" alt="R-CMD-check"></a></li>
<li><a href="https://codecov.io/github/krisrs1128/multimedia" class="external-link"><img src="https://codecov.io/github/krisrs1128/multimedia/graph/badge.svg?token=XB865GLQ8B" alt="codecov"></a></li>
<li><a href="https://mybinder.org/v2/gh/krisrs1128/multimedia/main" class="external-link"><img src="https://mybinder.org/badge_logo.svg" alt="Launch binder"></a></li>
</ul>
</div>

  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Kris Sankaran, Hanying Jiang.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
